### PR TITLE
PHP 8 support

### PR DESCRIPTION
**Pull request by sidz**
This PR:

- [x] adds PHP 8 support.
- [x] update dev dependencies

### CODE REVIEW

**Comment by ctrl-f5** [comment]
hello @sidz can you give this one an update? travis is failing but your previous MR moved builds to github actions, right?

**Comment by sidz** [comment]
@ctrl-f5

> travis is failing but your previous MR moved builds to github actions, right?

Yes. you are right. I'll rebase this PR.

> can you give this one an update?
I'll try to find some time to finish this PR. There also some issues with e2e tests.

File: composer.json [hunk 1]
```diff
@@ -39,19 +39,19 @@
...
+        "phpstan/phpstan-phpunit": "^0.12.17",
+        "phpstan/phpstan-strict-rules": "^0.12.9",
+        "phpstan/phpstan-webmozart-assert": "^0.12.9",
+        "phpunit/phpunit": "^9.5",
+        "thecodingmachine/safe": "^1.3",
+        "vimeo/psalm": "^4.4"
     },
     "conflict": {
-        "codeception/codeception": "<3.1.1"
+        "codeception/codeception": "<4.1.9"
```
Comment (sidz) [inline composer.json:55]: bumped up codeception/codeception as it compatible with both 7.4 and 8.0

**Review by maks-rafalko — APPROVED** [review]

**Comment by sidz** [comment]
Thanks @maks-rafalko 

@ctrl-f5 merged

**Pull request merged**